src/detect.js

```diff
@@ -18,15 +18,15 @@ function detectDeadCode(compilation, options) {
 
 function getPattern({ context, patterns, exclude }) {
   return patterns
-    .map((pattern) => path.resolve(context, pattern))
-    .concat(exclude.map((pattern) => `!${exclude}`));
+    .map(pattern => path.resolve(context, pattern))
+    .concat(exclude.map(pattern => `!${pattern}`));
 }
 
 function getUsedExportMap(compilation) {
   const unusedExportMap = {};
 
   compilation.chunks.forEach(function(chunk) {
-    chunk.forEachModule(function(module) {
+    for(const module of chunk.modulesIterable) {
```

Comment:
@sergivillar Is there any reason to make this change? Will it work for both webpack 3 and wepback 4 🤔